src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -367,17 +367,17 @@ protected override void OnContainersMaterialized(ItemContainerEventArgs e)
             {
                 if ((container.ContainerControl as ISelectable)?.IsSelected == true)
                 {
-                    if (SelectedIndex == -1)
-                    {
-                        SelectedIndex = container.Index;
-                    }
-                    else
+                    if (SelectionMode.HasFlag(SelectionMode.Multiple))
```

Comment:
It's not clear to me what this change to `SelectingItemsControl` fixes; could we have a failing/passing unit test in `SelectingItemsControlTests` that tests for this case?

Reply:
@grokys I have added a failing test. The issue only occurs with `AlwaysSelected` but still I think that materialized items should be able to always change selection so this fix was needed anyways.